[Profiler] Capture storage data pointer

robieta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #85162
* #85161
* __->__ #84276

This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

Commit:
[Profiler] Capture storage data pointer

This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

[ghstack-poisoned]

slgong-fb reviewed (approved):
LGTM. A lint issue exists.

Commit:
Update on "[Profiler] Capture storage data pointer"


This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Capture storage data pointer"


This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Capture storage data pointer"


This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Capture storage data pointer"


This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Capture storage data pointer"


This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Capture storage data pointer"


This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Capture storage data pointer"


This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Capture storage data pointer"


This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Capture storage data pointer"


This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Capture storage data pointer"


This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Capture storage data pointer"


This is approximately a re-land of the storage half of https://github.com/pytorch/pytorch/pull/80266

I've directly represented and exposed storage impl rather than using it as a first guess for an ID. (Mostly for testing, which happened to save me as I was initially recording the wrong thing.)

Differential Revision: [D39136546](https://our.internmc.facebook.com/intern/diff/D39136546/)

[ghstack-poisoned]
